File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
    int large=0;
        for(int i=0;i<arr.length-1;i++){
            int j=i+1;
            large=arr[j];
            while(j<arr.length-1){
                if(large<arr[j+1])
                    large=arr[j+1];
                j++;
            }
            arr[i]=large;
        }
        arr[arr.length-1]=-1;
        return arr;
    }
}

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.kt
class Solution {
    fun replaceElements(arr: IntArray): IntArray {
        
        if(arr.size==1){
            arr[0]=-1
            return arr
        }
        val length=arr.size
        var result=IntArray(length)
        result[length-1]=-1
        result[length-2]=arr[length-1]
        
        for(i in length-3 downTo 0){
            if(arr[i+1]>result[i+1]){
                result[i]=arr[i+1]
            }
            else{
                result[i]=result[i+1]
            }
        }
        return result
    }
}